src/elements.h

```diff
@@ -393,8 +393,14 @@ namespace libsemigroups {
     //! replaces the underlying vector of \c this with the underlying vector of
     //! \p x. Any method overriding this one must call
     //! Element::reset_hash_value.
+    //!
+    // TODO update the doc to reflect that this changes the degree of this if
+    // the degree of x is higher.
     void copy(Element const* x) override {
       LIBSEMIGROUPS_ASSERT(x->degree() == this->degree());
```

Comment:
This assertion directly contradicts the next line, so this will have one behaviour (assertion failure) when debugging is enabled, and another one (increasing the size) when it is not. There must not be any tests for this, because otherwise the CI would not have passed. I suggest deleting this change completely.